src/maxblocksize.cpp

```diff
@@ -0,0 +1,149 @@
+    }
+    else
+    {
+        uint64_t lowerFloor = nMaxBlockSize * 100 / 105;
+        lowerValue = (lowerValue < lowerFloor ? lowerFloor : lowerValue);
+        if (lowerValue < nMaxBlockSize)
+            nMaxBlockSize = lowerValue;
+    }
+
+    if (nMaxBlockSize != pindexLast->nMaxBlockSize)
+    {
+        LogPrintf("GetNextMaxBlockSize RETARGET\n");
+        LogPrintf("Before: %d\n", pindexLast->nMaxBlockSize);
+        LogPrintf("After:  %d\n", nMaxBlockSize);
+    }
+
+    return nMaxBlockSize;
+}
+
+static uint32_t FindVote(const std::string &coinbase)
+{
+    uint32_t eb_vote = 0;
+    std::vector<char> curr;
+    bool bip100vote = false;
+    bool started = false;
+
+    for (char s : coinbase)
+    {
+        if (s == '/')
+        {
+            started = true;
+            // End (or beginning) of a potential vote string.
+
+            if (curr.size() < 2) // Minimum vote string length is 2
+            {
+                bip100vote = false;
+                curr.clear();
+                continue;
+            }
+
+            if (std::string(begin(curr), end(curr)) == "BIP100")
+            {
+                bip100vote = true;
+                curr.clear();
+                continue;
+            }
+
+            // Look for a B vote.
+            if (bip100vote && curr[0] == 'B')
+            {
+                try
+                {
+                    return boost::lexical_cast<uint32_t>(std::string(begin(curr) + 1, end(curr)));
+                }
+                catch (const std::exception &e)
+                {
+                    LogPrintf("Invalid coinbase B-vote: %s\n", e.what());
+                }
+            }
+
+            // Look for a EB vote. Keep it, but continue to look for a BIP100/B vote.
+            if (!eb_vote && curr[0] == 'E' && curr[1] == 'B')
+            {
+                try
+                {
+                    eb_vote = boost::lexical_cast<uint32_t>(std::string(begin(curr) + 2, end(curr)));
+                }
+                catch (const std::exception &e)
+                {
+                    LogPrintf("Invalid coinbase EB-vote: %s\n", e.what());
+                }
+            }
+
+            bip100vote = false;
+            curr.clear();
+            continue;
+        }
+        else if (!started)
+            continue;
+        else
+            curr.push_back(s);
+    }
+    return eb_vote;
+}
+
+uint64_t GetMaxBlockSizeVote(const CScript &coinbase, int32_t nHeight)
+{
+    // Skip encoded height if found at start of coinbase
+    CScript expect = CScript() << nHeight;
+    int searchStart = coinbase.size() >= expect.size() && std::equal(expect.begin(), expect.end(), coinbase.begin()) ?
+                          expect.size() :
+                          0;
+
+    std::string s(coinbase.begin() + searchStart, coinbase.end());
+
+    if (s.length() < 5) // shortest vote is /EB1/
+        return 0;
+
+
+    return static_cast<uint64_t>(FindVote(s)) * 1000000;
```

Comment:
Ok, in thinking about it more I am not that happy with rounding /EB1.9/ down because the block size at a 1.05 maximum increase is definitely going to be ticking up in chunks that are less than 1MB.  So the miner needs to understand that 2 precisions are being used for the same value.  But ok, I see how it simplifies the code.   

But I think that interpreting /EB3.5/ as no vote (and therefore an implicit vote for the current limit) is a big mistake.  Large block votes may be highly underrepresented.  I think we need to round this down to the nearest whole number (i.e. 3).  This change is not that bad -- it remains limited to FindVote and since we are rounding down perfect representation does not matter.  So we can lexical cast to a float and then round down to a uint64_t.